cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -145,14 +156,32 @@ impl Cmd {
                     return Err(Error::Exit(status));
                 }
 
+                let file = format!("{}.wasm", p.name.replace('-', "_"));
+                let target_file_path = Path::new(target_dir)
+                    .join("wasm32-unknown-unknown")
+                    .join(&self.profile)
+                    .join(&file);
+
+                let mut wasm_bytes = fs::read(&target_file_path).unwrap();
+
+                self.metadata.iter().for_each(|(k, v)| {
+                    println!("Metadata: {}={}", k, v);
+                    let key: StringM = k.try_into().unwrap();
+                    let val: StringM = v.try_into().unwrap();
+                    let meta_entry = ScMetaEntry::ScMetaV0(ScMetaV0 { key, val });
+                    let xdr: Vec<u8> = meta_entry.to_xdr(Limits::none()).unwrap();
+
+                    wasm_gen::write_custom_section(&mut wasm_bytes, "contractmetav0", &xdr);
+                });
+
+                let updated_spec = Spec::new(&wasm_bytes).unwrap();
+                println!("Updated meta!: {:?}", updated_spec.meta);
+
+                let str_path: &str = target_file_path.to_str().unwrap();
+                fs::write(str_path, wasm_bytes).unwrap();
```

Comment:
You don't need to convert to a string. See the signature:
`pub fn write<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) -> Result<()> `
```suggestion
                fs::write(&target_file_path, wasm_bytes).unwrap();
```
Also should use an Error.